flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentControllerTest.java

```diff
@@ -274,6 +278,89 @@ public void verifyInProgressDeploymentWithCrashLoopBackoff() throws Exception {
         validatingResponseProvider.assertValidated();
     }
 
+    @ParameterizedTest
+    @EnumSource(FlinkVersion.class)
+    public void cleanUpDeploymentWithError(FlinkVersion flinkVersion)
+            throws JsonProcessingException {
```

Comment:
I think this test case is bit of an overkill.
To reproduce the logic and validate the fix it could simply be replaced with the following 2 lines:

```
        FlinkDeployment flinkDeployment = TestUtils.buildApplicationCluster();
        testController.cleanup(flinkDeployment, context);
```

We don't really need the error state, json etc because thats not related to the actual issue at hand I think. Also the rest of the method tests deployments, upgrades etc that are already tested in other methods in this class.

Reply:
ok